### PR TITLE
Map number(integer) into an integer rather than a float

### PR DESCRIPTION
**Pull request by rodireich**
## What
Closes https://github.com/airbytehq/oncall/issues/1228

On a mysql <> bigquery [connection](https://cloud.airbyte.io/workspaces/a5f2e623-3a04-4984-8705-4df95dda60c3/connections/fa46247e-fa0a-4f24-bbae-fcf7480a2fb9/status),
A src-mysql INT UNSIGNED column is transformed into an airbyte NUMBER type and then into a dst-bigquery FLOAT.
This confirms the [documented data type mapping](https://docs.airbyte.com/integrations/destinations/bigquery/#data-type-map) in bigquery.
But then something keeps failing on these number to float field during destination write.
Customer tried to disable a few of these failing tables but error keeps popping on similar ID columns in other tables.
I'm not sure why. Perhaps due to certain values.

This change will turn a source (number(integer) into a destination integer)
## How
Similarly to the way we convert a BigInt today, convert Integers into bigquery integers.

## 🚨 User Impact 🚨
This may affect existing connections. Need to validate before merging

Note: I applied the changes in #20640 on this PR to be able to run integration tests.

### CODE REVIEW

**Review by edgao — APPROVED** [review]

**Review by suhomud — APPROVED** [review]

**Review by ryankfu — APPROVED** [review]

File: airbyte-config/init/src/main/resources/seed/destination_definitions.yaml [hunk 1]
```diff
@@ -60,10 +60,6 @@
   dockerImageTag: 1.2.9
   documentationUrl: https://docs.airbyte.com/integrations/destinations/bigquery
   icon: bigquery.svg
-  normalizationConfig:
```
Comment (grishick) [inline airbyte-config/init/src/main/resources/seed/destination_definitions.yaml:63]: This PR https://github.com/airbytehq/airbyte/pull/20640 has fixed this problem. You may want to rebase these changes on master to keep the changes in this PR clean

File: airbyte-integrations/connectors/destination-bigquery-denormalized/src/main/java/io/airbyte/integrations/destination/bigquery/formatter/DefaultBigQueryDenormalizedRecordFormatter.java [hunk 2]
```diff
@@ -288,9 +289,11 @@ private static Builder getField(final StandardNameTransformer namingResolver, fi
             builder.setType(primaryType.getBigQueryType());
           }
           case NUMBER -> {
-            if (airbyteType != null && airbyteType.asText().equals("big_integer")) {
+            if (airbyteType != null
+                && StringUtils.equalsAnyIgnoreCase(airbyteType.asText(),
+                "big_integer", "integer")) {
               builder.setType(StandardSQLTypeName.INT64);
-            } else {
+            }  else {
```
Comment (grishick) [inline airbyte-integrations/connectors/destination-bigquery-denormalized/src/main/java/io/airbyte/integrations/destination/bigquery/formatter/DefaultBigQueryDenormalizedRecordFormatter.java:296]: nit: extra space

File: airbyte-integrations/connectors/destination-bigquery-denormalized/src/main/resources/spec.json [hunk 3]
```diff
@@ -2,7 +2,7 @@
   "documentationUrl": "https://docs.airbyte.com/integrations/destinations/bigquery",
   "supportsIncremental": true,
   "supportsNormalization": false,
-  "supportsDBT": true,
+  "supportsDBT": false,
```
Comment (grishick) [inline airbyte-integrations/connectors/destination-bigquery-denormalized/src/main/resources/spec.json:5]: same as my first comment - this change should not be part of this PR, so please rebase on master

**Review by grishick — COMMENTED** [review]
left a couple of comments. LGTM otherwise

File: airbyte-config/init/src/main/resources/seed/destination_definitions.yaml (hunk 1 above)
Comment (rodireich) [reply airbyte-config/init/src/main/resources/seed/destination_definitions.yaml:63]: OK. I see it was merged yesterday.
Updating

File: airbyte-integrations/connectors/destination-bigquery-denormalized/src/main/java/io/airbyte/integrations/destination/bigquery/formatter/DefaultBigQueryDenormalizedRecordFormatter.java (hunk 2 above)
Comment (rodireich) [reply airbyte-integrations/connectors/destination-bigquery-denormalized/src/main/java/io/airbyte/integrations/destination/bigquery/formatter/DefaultBigQueryDenormalizedRecordFormatter.java:296]: thanks!

**Review by grishick — APPROVED** [review]

**Review by jbfbell — APPROVED** [review]

**Comment by rodireich** [comment]
@grishick and the rest of destinations team: Should I go ahead with merging?

**Comment by edgao** [comment]
:shipit:

**Comment by rodireich** [comment]
/publish connector=connectors/destination-bigquery-denormalized
